src/util/signal_catcher.cpp

```diff
@@ -11,21 +11,27 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "signal_catcher.h"
 
 #if defined(_WIN32)
-#include <process.h>
 #else
 #include <cstdlib>
-#include <csignal>
 #endif
 
-#include <vector>
-
 // Here we have an instance of an ugly global object.
 // It keeps track of any child processes that we'll kill
-// when we are told to terminate.
+// when we are told to terminate. "No child" is indicated by '0'.
 
 #ifdef _WIN32
 #else
-std::vector<pid_t> pids_of_children;
+pid_t child_pid = 0;
+
+void register_child(pid_t pid)
+{
```

Comment:
We might want to include an invariant ensuring that only one child is registered.
If multiple child processes are actually supported here, then some kind of warning would be useful.